core/domain/topic_services.py

```diff
@@ -285,8 +285,8 @@ def apply_change_list(topic_id, change_list):
             elif change.cmd == topic_domain.CMD_DELETE_ADDITIONAL_STORY:
                 topic.delete_additional_story(change.story_id)
             elif change.cmd == topic_domain.CMD_ADD_UNCATEGORIZED_SKILL_ID:
-                _add_uncategorized_skill_id_to_topic(
-                    topic, change.new_uncategorized_skill_id)
+                topic.add_uncategorized_skill_id(
+                    change.new_uncategorized_skill_id)
```

Comment:
are we validating anywhere whether this `new_uncategorized_skill_id` corresponds to an existing skill?

Reply:
No, I don't think so, but from the frontend, this won't happen in normal use since this is only called when either a skill is unassigned from a subtopic or an existing skill is assigned to a topic.
That said, a validation might be required when the skill suggestion flow is added. Should I create an issue for it?

Reply:
@aks681 by "skill suggestion", you mean the question suggestion, right? If yes, then I'm not sure how the question suggestion workflow will affect this.

> but from the frontend, this won't happen in normal

The urls are working as API and we should all the data which comes through the request irrespective of the frontend behavior.

>  Should I create an issue for it?
Create an issue or fix it in the #8363 if it will be a small change.